ruma-events/src/enums.rs

```diff
@@ -149,11 +172,21 @@ impl<'de> de::Deserialize<'de> for AnyRoomEvent {
         D: de::Deserializer<'de>,
     {
         let json = Box::<RawJsonValue>::deserialize(deserializer)?;
-        let EventDeHelper { state_key, .. } = from_raw_json_value(&json)?;
+        let EventDeHelper { state_key, unsigned, .. } = from_raw_json_value(&json)?;
 
         if state_key.is_some() {
+            if let Some(unsigned) = unsigned {
+                if unsigned.redacted_because.is_some() {
+                    return Ok(AnyRoomEvent::RedactedState(from_raw_json_value(&json)?));
+                }
+            }
             Ok(AnyRoomEvent::State(from_raw_json_value(&json)?))
```

Comment:
These can all be made a bit more concise:

```suggestion
            Ok(match unsigned {
                Some(unsigned) if unsigned.redacted_because.is_some() => {
                    AnyRoomEvent::RedactedState(from_raw_json_value(&json)?)
                }
                _ => AnyRoomEvent::State(from_raw_json_value(&json)?)
            })
```